charts/zeebe-cluster-helm/test/statefulset_test.go

```diff
@@ -0,0 +1,55 @@
+package test
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+type statefulSetTemplateTest struct {
+	suite.Suite
+	chartPath string
+	release   string
+	namespace string
+	templates []string
+}
+
+func TestStatefulSetTemplate(t *testing.T) {
+	t.Parallel()
+
+	chartPath, err := filepath.Abs("../")
+	require.NoError(t, err)
+
+	suite.Run(t, &statefulSetTemplateTest{
+		chartPath: chartPath,
+		release:   "zeebe-cluster-helm",
+		namespace: "zeebe-" + strings.ToLower(random.UniqueId()),
+		templates: []string{"templates/statefulset.yaml"},
+	})
+}
+
+func (s *statefulSetTemplateTest) TestContainerSpecImage() {
```

Comment:
:thought_balloon: I'm not sure whether we want to do like that and write a ton of such tests or whether we write one test which sets (different to default) and enables everything. 

This test then compares against a golden file :thinking: I know golden file tests are also not optimal since they actually fail always, but at least with that we could also immediately see in the review process what is the actual change after applying the templates. :shrug: Any thoughts? 

Maybe @Sijoma has an opinion since they use also golden files in the controller. How do you like it ? :sweat_smile:

Reply:
I added a TOTAL simple example of it here https://github.com/camunda-community-hub/camunda-cloud-helm/commits/zell-test-goldenfile

If we change now for example a config then the test script could show something like:

Reply:
Yeah, the test itself isn't very useful, it was just an example of a template/unit test :smile: I think these kind of tests may be useful as you will get much faster feedback when you mess up configuration, versus deploying a whole installation. Plus, the whole point of the Helm charts _is_ to ease the configuration of the deployment, so we want to test that the configuration is correct, I think - and these types of tests are cheap and let you do it easily. I think it's worth exploring, we'd just have to figure out what scope we want to test (e.g. maybe not the standard variables like image.tag, but maybe just the ones we define?)

Reply:
That output with the exclamation marks and formatting is quite nice. I really like golden-files for configuration tests. It is easy to mess something up without them. 

One thing that is difficult with the controller at the moment is to properly scope when you need a golden file for some functionality. In our testing scenario its a bit easier to implement something that does not get picked up by the golden-files as we mock the reconcile function in our tests. I would assume that this is a bit easier with helm charts.